src/test/java/be/seeseemelk/mockbukkit/scheduler/BukkitSchedulerMockTest.java

```diff
@@ -179,4 +180,91 @@ void runTaskTimerAsynchronously_TaskExecutedOnSeperateThread() throws Interrupte
 		assertFalse(scheduler.isQueued(testTask.getTaskId()));
 		assertEquals(2, count.get());
 	}
+
+	@Test
+	public void getPendingTasks_Sync()
+	{
+		assertEquals(0, scheduler.getPendingTasks().size());
+		int amountTasks = 20;
+		AtomicInteger count = new AtomicInteger(amountTasks);
+		Runnable callback = count::decrementAndGet;
+		for (int i = 0; i < amountTasks; i++)
+		{
+			scheduler.runTaskLater(null, callback, 2L+(i%5));
+		}
+		while (count.get()>0)
+		{
+			assertEquals(count.get(), scheduler.getPendingTasks().size());
+			scheduler.performOneTick();
+		}
+		assertEquals(count.get(), scheduler.getPendingTasks().size());
+	}
+
+	@Test
+	public void getPendingTasks_Async()
+	{
+		assertEquals(0, scheduler.getPendingTasks().size());
+		int amountTasks = 20;
+		AtomicInteger count = new AtomicInteger(amountTasks);
+		Runnable callback = () ->
+		{
+			try
+			{	// simulate some varying work load / execution time
+				Thread.sleep(ThreadLocalRandom.current().nextInt(2, 20));
+			}
+			catch (InterruptedException e)
+			{
+				e.printStackTrace();
+			}
+			count.decrementAndGet();
+		};
+		for (int i = 0; i < amountTasks; i++)
+		{
+			scheduler.runTaskLaterAsynchronously(null, callback, 2L + (i % 5));
+		}
+		int pendingTasks;
+		int oldPendingTasks = amountTasks;
+		while ((pendingTasks = scheduler.getPendingTasks().size()) > 0)
+		{
+			int currentCount = count.get();
+			assertTrue(currentCount <= pendingTasks);
+			assertTrue(pendingTasks <= oldPendingTasks);
+			scheduler.performOneTick();
+			oldPendingTasks = pendingTasks;
+		}
+		assertEquals(0, scheduler.getPendingTasks().size());
+		assertEquals(0, count.get());
+	}
+
+	@Test
+	public void getPendingTasks_AsyncCanceledTasks()
+	{
+		assertEquals(0, scheduler.getPendingTasks().size());
+		int amountTasks = 20;
+		Runnable callback = () ->
+		{
+			try
+			{	// simulate some varying work load / execution time
+				Thread.sleep(ThreadLocalRandom.current().nextInt(2, 20));
+			}
+			catch (InterruptedException e){
+				e.printStackTrace();
+			}
+		};
+		for (int i = 0; i < amountTasks; i++)
+		{
+			scheduler.runTaskLaterAsynchronously(null, callback, 1L + (i % 5));
+		}
+		assertEquals(amountTasks, scheduler.getPendingTasks().size());
+		scheduler.performOneTick();
+		assertTrue(amountTasks >= scheduler.getPendingTasks().size());
+		assertNotEquals(0, scheduler.getPendingTasks().size());
```

Comment:
This might be better from a semantical standpoint.

```suggestion
		assertFalse(scheduler.getPendingTasks().isEmpty());
```